amp_conf/htdocs/admin/libraries/BMO/Ajax.class.php

```diff
@@ -83,6 +84,12 @@ public function doRequest($module = null, $command = null) {
 		}
 
 		session_start();
+
+                // If this is a localhost request (ie this server has sent an ajax request to this server) then
+                // no need to authenticate.
+                if ($url['host'] == '127.0.0.1') $this->settings['authenticate'] = false;
```

Comment:
Referrer can be spoofed, but because of the previous check you will only get here both referrer and host are the same, so this check then requires that both host and referrer are 127.0.0.1. The second is easy to spoof, but the first? And by checking referrer, we add a little twist in, by requiring the request to be well formed both for host and referrer.